modules/tinymce/src/core/main/ts/newline/InsertNewLine.ts

```diff
@@ -9,21 +9,39 @@ import { Fun, Type } from '@ephox/katamari';
 
 import Editor from '../api/Editor';
 import { EditorEvent } from '../api/util/EventDispatcher';
-import { fireFakeInputEvent } from '../keyboard/FakeInputEvents';
+import { fireFakeBeforeInputEvent, fireFakeInputEvent } from '../keyboard/FakeInputEvents';
 import * as InsertBlock from './InsertBlock';
 import * as InsertBr from './InsertBr';
 import * as NewLineAction from './NewLineAction';
 
 const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>) => {
   NewLineAction.getAction(editor, evt).fold(
     () => {
+      if (Type.isNonNullable(evt)) {
+        const event = fireFakeBeforeInputEvent(editor, 'insertLineBreak');
+
+        if (event.isDefaultPrevented()) {
```

Comment:
```suggestion
        const event = fireFakeBeforeInputEvent(editor, 'insertLineBreak');
        if (event.isDefaultPrevented()) {
```
This is a little too verbose/spaced out and as such makes it harder to read, especially given this is all one logical unit.